### ISSUE TITLE
[1.2.0] test failure with PHP 8.0 and TS build

### ISSUE THREAD

**Issue by remicollet** [scoutapp/scout-apm-php-ext#67]
PHP 7.x are OK
NTS build are OK

```
=====================================================================
PHP         : /usr/bin/zts-php 
PHP_SAPI    : cli
PHP_VERSION : 8.0.2
ZEND_VERSION: 4.0.2
PHP_OS      : Linux - Linux builder.remirepo.net 5.10.11-200.fc33.x86_64 #1 SMP Wed Jan 27 20:21:22 UTC 2021 x86_64
INI actual  : /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS
More .INIs  :   
CWD         : /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS
Extra dirs  : 
VALGRIND    : Not used
=====================================================================
TIME START 2021-02-04 13:00:08
=====================================================================
TEST 1/18 [tests/001-check-ext-loaded.phpt]
PASS Check Scout APM extension is loaded [tests/001-check-ext-loaded.phpt] 
TEST 2/18 [tests/002-file_get_contents.phpt]
========DIFF========
001+ Warning: Undefined array key 0 in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/002-file_get_contents.php on line 3
002+ 
003+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/002-file_get_contents.php on line 5
004+ NULL
005+ 
006+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/002-file_get_contents.php on line 6
007+ NULL
008+ 
009+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/002-file_get_contents.php on line 7
001- string(17) "file_get_contents"
002- float(%f)
003- float(%f)
004- float(%f)
005- bool(true)
006- array(1) {
007-   [0]=>
008-   string(%d) "%s/tests/002-file_get_contents.php"
009- }
010+ NULL
011+ 
012+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/002-file_get_contents.php on line 8
013+ NULL
014+ 
015+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/002-file_get_contents.php on line 9
016+ 
017+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/002-file_get_contents.php on line 9
018+ bool(false)
019+ 
020+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/002-file_get_contents.php on line 10
021+ NULL
========DONE========
FAIL Calls to file_get_contents are logged [tests/002-file_get_contents.phpt] 
TEST 3/18 [tests/003-scoutapm_get_calls-clears-calls-list.phpt]
========DIFF========
001+ int(0)
002+ int(0)
001- int(1)
     int(0)
003- int(1)
     int(0)
========DONE========
FAIL Ensures that calls to scoutapm_get_calls() clears the call list [tests/003-scoutapm_get_calls-clears-calls-list.phpt] 
TEST 4/18 [tests/004-namespaced-fgc-is-not-logged.phpt]
========DIFF========
--
     array(0) {
     }
     SHOULD be logged
011+ array(0) {
012+ }
011- array(1) {
012-   [0]=>
013-   array(5) {
014-     ["function"]=>
015-     string(17) "file_get_contents"
016-     ["entered"]=>
017-     float(%f)
018-     ["exited"]=>
019-     float(%f)
020-     ["time_taken"]=>
021-     float(%f)
022-     ["argv"]=>
023-     array(1) {
024-       [0]=>
025-       string(%d) "%s/tests/004-namespaced-fgc-is-not-logged.php"
026-     }
027-   }
028- }
========DONE========
FAIL Calls to file_get_contents are logged [tests/004-namespaced-fgc-is-not-logged.phpt] 
TEST 5/18 [tests/005-requiring-external-files-handled.phpt]
PASS Requires to external files do not crash [tests/005-requiring-external-files-handled.phpt] 
TEST 6/18 [tests/006-anonymous-classes-handled.phpt]
PASS Executing a closure/anonymous function does not crash [tests/006-anonymous-classes-handled.phpt] 
TEST 7/18 [tests/007-evaled-code-handled.phpt]
PASS Running evaled code does not crash [tests/007-evaled-code-handled.phpt] 
TEST 8/18 [tests/008-class-with-no-constructor-call-handled.phpt]
PASS Class without a defined constructor does not crash [tests/008-class-with-no-constructor-call-handled.phpt] 
TEST 9/18 [tests/009-curl_exec.phpt]
SKIP Calls to curl_exec are logged [tests/009-curl_exec.phpt] reason: scout was not compiled with curl
TEST 10/18 [tests/010-fwrite-fread-fopen.phpt]
========DIFF========
     fread/fwrite test
002+ 
003+ Warning: Undefined array key 0 in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/010-fwrite-fread-fopen.php on line 11
004+ 
005+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/010-fwrite-fread-fopen.php on line 11
006+ NULL
007+ 
008+ Warning: Undefined array key 0 in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/010-fwrite-fread-fopen.php on line 12
009+ 
010+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/010-fwrite-fread-fopen.php on line 12
011+ NULL
012+ 
013+ Warning: Undefined array key 1 in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/010-fwrite-fread-fopen.php on line 14
014+ 
015+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/010-fwrite-fread-fopen.php on line 14
002- string(6) "fwrite"
003- array(2) {
004-   [0]=>
005-   string(%d) "/tmp/scoutapm-test%s"
006-   [1]=>
007-   string(%d) "w+"
008- }
009- string(5) "fread"
010- array(2) {
011-   [0]=>
012-   string(%d) "/tmp/scoutapm-test%s"
013-   [1]=>
014-   string(%d) "w+"
015- }
016+ NULL
017+ 
018+ Warning: Undefined array key 1 in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/010-fwrite-fread-fopen.php on line 15
019+ 
020+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/010-fwrite-fread-fopen.php on line 15
021+ NULL
========DONE========
FAIL Calls to fwrite and fread are logged with handle from fopen() [tests/010-fwrite-fread-fopen.phpt] 
TEST 11/18 [tests/010-fwrite-fread-tmpfile.phpt]
========DIFF========
     fread/fwrite test
002+ 
003+ Warning: Undefined array key 0 in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/010-fwrite-fread-tmpfile.php on line 11
004+ 
005+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/010-fwrite-fread-tmpfile.php on line 11
006+ NULL
007+ 
008+ Warning: Undefined array key 0 in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/010-fwrite-fread-tmpfile.php on line 12
009+ 
010+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/010-fwrite-fread-tmpfile.php on line 12
011+ NULL
012+ 
013+ Warning: Undefined array key 1 in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/010-fwrite-fread-tmpfile.php on line 14
014+ 
015+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/010-fwrite-fread-tmpfile.php on line 14
016+ NULL
002- string(6) "fwrite"
003- array(2) {
004-   [0]=>
005-   resource(%d) of type (Unknown)
006-   [1]=>
007-   string(18) "fread/fwrite test
008- "
009- }
010- string(5) "fread"
011- array(2) {
012-   [0]=>
013-   resource(%d) of type (Unknown)
014-   [1]=>
015-   int(18)
016- }
017+ 
018+ Warning: Undefined array key 1 in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/010-fwrite-fread-tmpfile.php on line 15
019+ 
020+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/010-fwrite-fread-tmpfile.php on line 15
021+ NULL
========DONE========
FAIL Calls to fwrite and fread are logged with handle from tmpfile() [tests/010-fwrite-fread-tmpfile.phpt] 
TEST 12/18 [tests/011-pdo-exec.phpt]
========DIFF========
001+ Warning: Undefined array key 0 in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/011-pdo-exec.php on line 7
002+ 
003+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/011-pdo-exec.php on line 7
004+ NULL
001- string(9) "PDO->exec"
002- string(38) "CREATE TABLE foo (col INT PRIMARY KEY)"
003- string(9) "PDO->exec"
004- string(38) "INSERT INTO foo (col) VALUES (1), (2) "
005+ 
006+ Warning: Undefined array key 0 in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/011-pdo-exec.php on line 8
007+ 
008+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/011-pdo-exec.php on line 8
009+ 
010+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/011-pdo-exec.php on line 8
011+ NULL
012+ 
013+ Warning: Undefined array key 1 in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/011-pdo-exec.php on line 9
014+ 
015+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/011-pdo-exec.php on line 9
016+ NULL
017+ 
018+ Warning: Undefined array key 1 in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/011-pdo-exec.php on line 10
019+ 
020+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/011-pdo-exec.php on line 10
021+ 
022+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/011-pdo-exec.php on line 10
023+ NULL
========DONE========
FAIL Calls to PDO::exec are logged [tests/011-pdo-exec.phpt] 
TEST 13/18 [tests/011-pdo-query.phpt]
========DIFF========
--
       ["1 + 2"]=>
       string(1) "3"
     }
005+ 
006+ Warning: Undefined array key 0 in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/011-pdo-query.php on line 7
005- string(10) "PDO->query"
006- string(12) "SELECT 1 + 2"
007+ 
008+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/011-pdo-query.php on line 7
009+ NULL
010+ 
011+ Warning: Undefined array key 0 in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/011-pdo-query.php on line 8
012+ 
013+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/011-pdo-query.php on line 8
014+ 
015+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/011-pdo-query.php on line 8
016+ NULL
========DONE========
FAIL Calls to PDO::query are logged [tests/011-pdo-query.phpt] 
TEST 14/18 [tests/011-pdostatement-execute-pdo-prepare.phpt]
========DIFF========
001+ Warning: Undefined array key 0 in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/011-pdostatement-execute-pdo-prepare.php on line 9
002+ 
003+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/011-pdostatement-execute-pdo-prepare.php on line 9
004+ NULL
005+ 
006+ Warning: Undefined array key 0 in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/011-pdostatement-execute-pdo-prepare.php on line 10
007+ 
008+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/011-pdostatement-execute-pdo-prepare.php on line 10
009+ NULL
010+ 
001- string(21) "PDOStatement->execute"
002- array(1) {
003-   [0]=>
004-   string(%d) "SELECT 1 + 2"
005- }
006- string(21) "PDOStatement->execute"
007- array(1) {
008-   [0]=>
009-   string(%d) "SELECT 3 + 4"
010- }
011+ Warning: Undefined array key 1 in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/011-pdostatement-execute-pdo-prepare.php on line 11
012+ 
013+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/011-pdostatement-execute-pdo-prepare.php on line 11
014+ NULL
015+ 
016+ Warning: Undefined array key 1 in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/011-pdostatement-execute-pdo-prepare.php on line 12
017+ 
018+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/011-pdostatement-execute-pdo-prepare.php on line 12
019+ NULL
========DONE========
FAIL Calls to PDOStatement::execute are logged when created from PDO->prepare() [tests/011-pdostatement-execute-pdo-prepare.phpt] 
TEST 15/18 [tests/012-file_put_contents.phpt]
========DIFF========
001+ Warning: Undefined array key 0 in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/012-file_put_contents.php on line 4
     string(12) "test content"
003+ 
004+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/012-file_put_contents.php on line 7
005+ NULL
006+ 
007+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/012-file_put_contents.php on line 8
008+ NULL
002- string(17) "file_put_contents"
003- array(2) {
004-   [0]=>
005-   string(%d) "%sscout-apm-php-ext-test%s"
006-   [1]=>
007-   string(12) "test content"
008- }
========DONE========
FAIL Calls to file_put_contents are logged [tests/012-file_put_contents.phpt] 
TEST 16/18 [tests/013-fix-memory-leak-when-scoutapm_get_calls-not-called.phpt]
SKIP When calls to functions are recorded, and scoutapm_get_calls is not called, don't leak memory [tests/013-fix-memory-leak-when-scoutapm_get_calls-not-called.phpt] reason: Recompile PHP with --enable-debug.
TEST 17/18 [tests/bug-47.phpt]
========DIFF========
001+ Warning: Undefined array key 0 in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/bug-47.php on line 6
002+ 
003+ Warning: Trying to access array offset on value of type null in /builddir/build/BUILD/php-pecl-scoutapm-1.2.0/ZTS/tests/bug-47.php on line 6
004+ NULL
001- array(2) {
002-   [0]=>
003-   resource(%d) of type (%s)
004-   [1]=>
005-   string(%d) "fread/fwrite test"
006- }
========DONE========
FAIL Bug https://github.com/scoutapp/scout-apm-php-ext/issues/47 - fix segfault when accessing argument store out of bounds [tests/bug-47.phpt] 
TEST 18/18 [tests/bug-49.phpt]
PASS Bug https://github.com/scoutapp/scout-apm-php-ext/issues/49 - only record arguments for fopen if it returns a resource [tests/bug-49.phpt] 
=====================================================================
TIME END 2021-02-04 13:00:09
=====================================================================
TEST RESULT SUMMARY
---------------------------------------------------------------------
Exts skipped    :    0
Exts tested     :   19
---------------------------------------------------------------------
Number of tests :   18                16
Tests skipped   :    2 ( 11.1%) --------
Tests warned    :    0 (  0.0%) (  0.0%)
Tests failed    :   10 ( 55.6%) ( 62.5%)
Tests passed    :    6 ( 33.3%) ( 37.5%)
---------------------------------------------------------------------
Time taken      :    1 seconds
=====================================================================
=====================================================================
FAILED TEST SUMMARY
---------------------------------------------------------------------
Calls to file_get_contents are logged [tests/002-file_get_contents.phpt]
Ensures that calls to scoutapm_get_calls() clears the call list [tests/003-scoutapm_get_calls-clears-calls-list.phpt]
Calls to file_get_contents are logged [tests/004-namespaced-fgc-is-not-logged.phpt]
Calls to fwrite and fread are logged with handle from fopen() [tests/010-fwrite-fread-fopen.phpt]
Calls to fwrite and fread are logged with handle from tmpfile() [tests/010-fwrite-fread-tmpfile.phpt]
Calls to PDO::exec are logged [tests/011-pdo-exec.phpt]
Calls to PDO::query are logged [tests/011-pdo-query.phpt]
Calls to PDOStatement::execute are logged when created from PDO->prepare() [tests/011-pdostatement-execute-pdo-prepare.phpt]
Calls to file_put_contents are logged [tests/012-file_put_contents.phpt]
Bug https://github.com/scoutapp/scout-apm-php-ext/issues/47 - fix segfault when accessing argument store out of bounds [tests/bug-47.phpt]
=====================================================================

```

**Comment by remicollet** [comment]
Strangely passes on RHEL-8 but fails on RHEL-7 and Fedora 31 to 33

**Comment by asgrim** [comment]
Passes on the build matrix on CI (Ubuntu Bionic), but not on my local machine. Very odd. Am trying to work on a fix (see https://github.com/scoutapp/scout-apm-php-ext/pull/68#discussion_r570335740 ) but it doesn't seem the right fix, so need to have a think/investigate more. Alternatively, if you have any feedback/thoughts, also welcome!

**Comment by asgrim** [comment]
Fixed in #69
